File: unity/Assets/Scripts/Multiplayer/Matchmaking/MatchmakingCurrentResponse.cs
using Michitai.Multiplayer.Errors;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Michitai.Multiplayer.Matchmaking
{
    /// <summary>
    /// Response containing the current matchmaking status for a player.
    /// Includes lobby information and pending join requests.
    /// </summary>
    /// <typeparam name="T">The type to deserialize lobby rules into.</typeparam>
    [System.Serializable]
    public class MatchmakingCurrentResponse<T> : ApiResponse<EMatchmakingCurrentError> where T : class, new()
    {
        /// <summary>
        /// Indicates whether the player is currently in a matchmaking lobby.
        /// </summary>
        public bool in_matchmaking;

        /// <summary>
        /// The matchmaking lobby information if the player is in one.
        /// </summary>
        public MatchmakingInfo<T> matchmaking;

        /// <summary>
        /// List of pending join requests (for host approval).
        /// </summary>
        public List<MatchmakingRequestBase> pending_requests;
    }
}

File: java/src/main/java/com/michitai/multiplayer/matchmaking/MatchmakingCurrentResponse.java
package com.michitai.multiplayer.matchmaking;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.JsonNode;
import com.michitai.multiplayer.ApiResponse;

import java.util.List;

/**
 * Response containing the current matchmaking status for a player.
 * Includes lobby information and pending join requests.
 *
 * @param <T> The type to deserialize lobby rules into.
 */
public class MatchmakingCurrentResponse<T> extends ApiResponse {
    @JsonProperty("in_matchmaking")
    private boolean inMatchmaking;

    @JsonProperty("matchmaking")
    private MatchmakingInfo<T> matchmaking;

    @JsonProperty("pending_requests")
    private List<MatchmakingRequestBase> pendingRequests;

    public boolean isInMatchmaking() {
        return inMatchmaking;
    }

    public void setInMatchmaking(boolean inMatchmaking) {
        this.inMatchmaking = inMatchmaking;
    }

    public MatchmakingInfo<T> getMatchmaking() {
        return matchmaking;
    }

    public void setMatchmaking(MatchmakingInfo<T> matchmaking) {
        this.matchmaking = matchmaking;
    }

    public List<MatchmakingRequestBase> getPendingRequests() {
        return pendingRequests;
    }

    public void setPendingRequests(List<MatchmakingRequestBase> pendingRequests) {
        this.pendingRequests = pendingRequests;
    }
}
